src/nvim/context.c

```diff
@@ -271,8 +272,10 @@ static inline void ctx_save_funcs(Context *ctx, bool scriptonly)
       size_t cmd_len = sizeof("func! ") + strlen(name);
       char *cmd = xmalloc(cmd_len);
       snprintf(cmd, cmd_len, "func! %s", name);
-      String func_body = nvim_exec(VIML_INTERNAL_CALL, cstr_as_string(cmd),
-                                   true, &err);
+      Dict(exec_opts) opts = { 0 };
+      opts.output = BOOLEAN_OBJ(true);
+      String func_body = exec_impl(VIML_INTERNAL_CALL, cstr_as_string(cmd),
```

Comment:
Why not call `nvim_exec2`, why do we need `exec_impl` ?

Reply:
Short answer: I tried but couldn't do it. This worked, so decided to use it.

Long answer: I didn't manage to write proper C code to get `output` field from `nvim_exec2()` dictionary output.

Code for `nvim_exec()` was like this:
```c
String nvim_exec(uint64_t channel_id, String src, Boolean output, Error *err)
  FUNC_API_SINCE(7)
  FUNC_API_DEPRECATED_SINCE(11)
{
  Dict(exec_opts) opts = { 0 };
  opts.output = BOOLEAN_OBJ(output);
  Dictionary exec2_output = nvim_exec2(channel_id, src, &opts, err);

  for (size_t i = 0; i < exec2_output.size; i++) {
    if (strequal(exec2_output.items[i].key.data, "output")) {
      return exec2_output.items[i].value.data.string;
    }
  }
  return (String)STRING_INIT;
}
```

I thought this should essentially be `return nvim_exec2(src, { output = output }).output` but it always returned empty string (while same `nvim_exec2()` call does return non-empty string). There seems to also be some potential memory leak, so Bjorn suggested to write a helper while leaving `nvim_exec2()` to construct output dictionary.